plugins/modules/locale_gen.py

```diff
@@ -90,14 +114,29 @@ class LocaleGen(StateModuleHelper):
     use_old_vardict = False
 
     def __init_module__(self):
-        self.vars.set("ubuntu_mode", False)
-        if os.path.exists(self.LOCALE_SUPPORTED):
+        self.MECHANISMS = dict(
+            ubuntu_legacy=dict(
+                available=SUPPORTED_LOCALES,
+                apply_change=self.apply_change_ubuntu_legacy,
+            ),
+            glibc=dict(
+                available=SUPPORTED_LOCALES,
+                apply_change=self.apply_change_glibc,
+            ),
+        )
+
+        if os.path.exists(ETC_LOCALE_GEN):
+            self.vars.ubuntu_mode = False
+            self.vars.mechanism = "glibc"
+        elif os.path.exists(VAR_LIB_LOCALES):
             self.vars.ubuntu_mode = True
+            self.vars.mechanism = "ubuntu_legacy"
```

Comment:
Maybe something like:
```suggestion
            self.vars.mechanism = "ubuntu_legacy"
            self.module.deprecate(
                "On this machine mechanism=ubuntu_legacy is used. This mechanism is deprecated and will be removed from"
                " in community.general 13.0.0. If you see this message on a modern Debian or Ubuntu version,"
                " please create an issue in the community.general repository",
                version="13.0.0", collection_name="community.general"
            )
```